src/ReadingPages.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { Recipe } from "./recipe";
+import { Image } from "antd";
+import "antd/dist/antd.min.css";
+import "./Reading.css";
+
+export default function ReadingPage() {
+  const emptyRecipe: Recipe = {
+    id: 0,
+    name: "0",
+    thumbnail: {
+      image: "",
+    },
+    description: "",
+    link: ["0"],
+    metadata: {
+      lastViewed: "0",
+      created: "0",
+      timeToCook: "0",
+    },
+    ingredients: [
+      {
+        ingredient: "",
+        starred: true,
+      },
+    ],
+    steps: [
+      {
+        step: 0,
+        directions: "",
+        image: "",
+      },
+    ],
+  };
+
+  const [readingRecipe, setReadingRecipe] = useState<Recipe>(emptyRecipe);
+  const [loading, setLoading] = useState(true);
+
+  const id = useParams().recipeId;
+  if (!id) {
+    throw new Error("id is undefined");
+  }
+
+  const baseUrl = `${process.env.REACT_APP_API_URL}`;
+  if (!baseUrl || baseUrl === "undefined") {
+    throw new Error("The URL environment variable is undefined/missing");
+  }
+
+  const readingUrl = baseUrl + "/recipe/" + id;
+
+  const getRecipe = async () => {
+    const response = await fetch(readingUrl);
+    const data: Recipe = await response.json();
+    if (!data) {
+      throw new Error(
+        "The fetch call did not return any data matching the Recipe Type"
+      );
+    }
+    setReadingRecipe(data);
+    setLoading(false);
+  };
+
+  const showImage = (imageSrc: string) => {
+    if (imageSrc !== "") {
```

Comment:
You have image marked as a non nullable string in your types.  Do you need to make it a nullable string?